plugins/azure/keyvaults/cmkCreationForAppTier.js

```diff
@@ -0,0 +1,86 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+var _ = require('underscore');
+
+module.exports = {
+    title: 'CMK Creation for App Tier Enabled',
+    category: 'Key Vaults',
+    domain: 'Identity and Access Management',
+    description: 'Ensure that a Customer-Managed Key (CMK) is created and configured for your Microsoft Azure application tier.',
+    more_info: 'Setting a CMK for app tier, you gain full control over who can use this key to access the application data, implementing the principle of least privilege on the encryption key ownership and usage.',
+    recommended_action: 'Ensure each Key Vault has a CMK created and configured.',
+    link: 'https://docs.microsoft.com/en-us/azure/azure-app-configuration/concept-customer-managed-keys',
+    apis: ['vaults:list', 'vaults:getKeys'],
+    settings: {
+        app_tier_tag_sets: {
+            name: 'App Tier Tag Sets',
+            description: 'An object of allowed tag set key value pairs to use for the CMKs creation for App Tier',
+            // eslint-disable-next-line no-useless-escape
+            regex: '/("?)\b(\w+)\1\s*:\s*("?)((?:\w+[-+*%])*?\w+)\b\3/g',
+            default: {}
+        }
+    },
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+        var config = {
+            app_tier_tag_sets: settings.app_tier_tag_sets || this.settings.app_tier_tag_sets.default
+        };
+
```

Comment:
Make this optional plugin i.e. if (!config.app_tier_tag_sets.length) return callback(null, results, source);